### DISCUSSION TITLE
Scalar function ends_with does not exist

### DISCUSSION THREAD

**Discussion by holdenmatt** [duckdb/duckdb#8757] [General]
### What happens?

Just reporting an inconsistency in text function naming.

I've been using `starts_with` for a while, so was surprised to find no `ends_with` function exists.

It looks like prefix/suffix do the same thing, but I find those names less intuitive (both python/JS call it endswith). Adding `ends_with` as an alias of `suffix` would be more consistent with `starts_with` and I think more intuitive for new users.

### To Reproduce

duckdb> select starts_with('foo', 'f');
true

duckdb> select ends_with('foo', 'f');
"Catalog Error: Scalar Function with name ends_with does not exist!"

### OS:

macOS

### DuckDB Version:

v0.8.2-dev1781

### DuckDB Client:

All

### Full Name:

Matt Holden

### Affiliation:

Duckbook.ai (solo dev)

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by holdenmatt** [comment]
(apologies if this isn't the right place for issues like this. I can move to Discussions if that's preferred)

**Comment by hannes** [comment]
Indeed this is more of a feature request. Should be rather straightforward to add an alias, happy to review a PR.

**Reply by holdenmatt** [reply]
Ok, I can look into sending a PR.

It looks like `starts_with` is implemented in `core_functions/scalar/string`, while `prefix`/`suffix` are in `function/scalar/string`.

Which do you prefer?
1. Add an `ends_with` implementation and tests alongside `starts_with`?
2. Alias `ends_with` to `suffix` (simpler / less code, but not consistent with how starts_with currently works)

No difference from a user perspective, so more a question of how you prefer the code to be organized.

**Reply by Tishj** [reply]
I assume adding an alias in `default_functions.cpp` would be sufficient

**Reply by holdenmatt** [reply]
Ah nice, even easier, thanks for the pointer!

**Reply by holdenmatt** [reply]
Looks like it's an easy one-line code change.

Should I submit a separate PR to the duckdb-web repo to add a line to the docs here?
https://github.com/duckdb/duckdb-web/blob/57d1d3861af9e5b8927696b6ddc8464f0100c0b5/docs/sql/functions/char.md?plain=1

(I see some docs are autogenerated, but afaict that one isn't?)

**Reply by Tishj** [reply]
Yes that's a great idea!

**Reply by holdenmatt** [reply]
FYI, submitted both PRs: [alias](https://github.com/duckdb/duckdb/pull/8763), [docs](https://github.com/duckdb/duckdb-web/pull/1071).
Thanks for considering!
